agent/api/task/task.go

```diff
@@ -846,6 +876,81 @@ func (task *Task) dockerHostConfig(container *apicontainer.Container, dockerCont
 	return hostConfig, nil
 }
 
+func (task *Task) awslogsEndpointOverride(hostConfig *docker.HostConfig, dockerVersion string) bool {
+	if hostConfig.LogConfig.Type != awslogsType {
+		return false
+	}
+
+	isVersionMatch := false
+	if task.isDockerCEVersion(dockerVersion) {
+		match, err := utils.Version(dockerVersion).Matches(">=" + awslogsEndpointMinimumDockerCEVersion)
+		if err != nil {
+			seelog.Warnf("Unable to compare docker version: %v", err)
+		} else if match {
+			isVersionMatch = true
+		}
+	} else if task.isDockerEEVersion(dockerVersion) {
+		match, err := utils.Version(dockerVersion).Matches(">=" + awslogsEndpointMinimumDockerEEVersion)
+		if err != nil {
+			seelog.Warnf("Unable to compare docker version: %v", err)
+		} else if match {
+			isVersionMatch = true
+		}
+	} else {
+		seelog.Warnf("Docker version %s is neither CE version or EE version", dockerVersion)
+	}
+
+	if isVersionMatch {
+		return hostConfig.LogConfig.Config[awslogsEndpointOpt] == ""
+	} else {
+		// If the Docker version is not matched, make the awslogs endpoint option empty
```

Comment:
You're now overriding something that is set either in the task definition or by the ECS backend.  I'd caution against this as logic here would prevent us from forcing an override if we need to.